File: leetcode121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        leftPointer = 0
        rightPointer = 1
        maxProfit = 0
        while rightPointer < len(prices):
            if prices[leftPointer] < prices[rightPointer]:
                maxProfit = max(prices[rightPointer] - prices[leftPointer], maxProfit)
            else:
                leftPointer = rightPointer
            rightPointer += 1
        return maxProfit

File: javaStuff/leetcode121.java
class Solution {
    public int maxProfit(int[] prices) {
        int leftPointer = 0;
        int rightPointer = 1;
        int maxProfit = 0;
        while(rightPointer < prices.length){
            if(prices[leftPointer] < prices[rightPointer])
                maxProfit = Math.max(prices[rightPointer] - prices[leftPointer], maxProfit);
            else leftPointer = rightPointer;
            rightPointer++;
        }
        return maxProfit;
    }
}